tutel/impls/moe_layer.py

```diff
@@ -112,7 +116,7 @@ def __init__(
         self.skip_moe = (int(os.environ.get('SKIP_MOE', '0')) != 0)
 
         self.num_local_experts = experts.pop('count_per_node', 1)
-        self.register_buffer('_num_global_experts', torch.tensor(MOELayer.global_expert_count(self.num_local_experts, self.group)))
+        self._num_global_experts = MOELayer.global_expert_count(self.num_local_experts, self.group)
 
```

Comment:
May I ask what is the difference with register_buffer()?

Reply:
Sharding libraries generally shard all network parameters and buffers
This is a constant which shouldn't be treated like a parameters or buffers

Reply:
Make sense, but seems like it is a global thing right now so that all layers use the same global_expert_count? And I also found few issues in some other changes where I am still investigating whether it has influence in existing features.

Reply:
yes all layers use the same global_expert_count
is there any reason for diff rank to output diff things from `MOELayer.global_expert_count`? 
just setting it as a python int seems like a valid approach
(Using a buffer results in fsdp treating it as part of the state which should be sharded, and this produces issues in some corner cases.)

Reply:
> I also found few issues in some other changes where I am still investigating whether it has influence in existing features.

👀 lmk what you find